src/celeritas/track/detail/StatusCheckExecutor.hh

```diff
@@ -0,0 +1,123 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/track/detail/StatusCheckExecutor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "celeritas/global/CoreTrackView.hh"
+#include "celeritas/track/SimTrackView.hh"
+
+#include "../StatusCheckData.hh"
+
+//! Check that the condition is true, otherwise throw an error/assertion
+#define SCE_ASSERT(COND, MSG)                                         \
```

Comment:
undef at the bottom of that file?

Reply:
Maybe I'll leave the macro with a `CELER_` prefix and additional documentation ... it's still definitely a "detail" but I could see it being used elsewhere...